pytorch_lightning/trainer/logging.py

```diff
@@ -57,7 +55,12 @@ def log_metrics(self, metrics, grad_norm_dic, step=None):
         # turn all tensors to scalars
         scalar_metrics = self.metrics_to_scalars(metrics)
 
-        step = step if step is not None else self.global_step
+        if "step" in scalar_metrics and step is None:
```

Comment:
I would write... `if not step and "step" in scalar_metrics:`

Reply:
My idea was not to break anything, if `step` is not None, then `scalar_metrics["step"]` is ignored: explicit is better than implicit. By using your code if `step=0` then `scalar_metrics["step"]` will actually be used. So could you explain why you'd prefer it in this way?

Reply:
well, we are on the path to having everything tested so beaking shall be revied via tests...
the notation `if step is not None` and `if step` because `step` is not `False`  nor `0`, right?
but let keep as it is now... :]

Reply:
`step` could be 0, as many people prefer to count from it. Checking for `None` is a safer way in my opinion.

Reply:
agree :]